src/service/data_products/common_functions.py

```diff
@@ -337,36 +340,31 @@ async def query_table(
     will be used to mark which rows are matched / selected by a value of `True`.
 
     storage - the storage system.
+    columns - the expected columns in the table data.
     filters - the filters to apply to the search
     output_table - whether to return the results as a list of lists (e.g. a table) with a separate
         fields entry defining the key for each table column, or a list of key / value dictionaries.
     document_mutator - a function applied to a document retrieved from the database before
         returning the results.
     """
+    fields = [c.key for c in columns]
```

Comment:
should we enforce `sort` here rather than relying on caller to pass a `sorted(cols)`?

Reply:
I'm assuming that in the future we'll probably want to define a sort order in the column specs, in which case the calling code will need to provide the sort order to `query_table`

Reply:
👍